plaidml/edsl/edsl.h

```diff
@@ -920,6 +924,10 @@ inline Tensor ident(const Tensor& x) { return intrinsic("ident", x); }
 /// \return Tensor
 ///
 inline Tensor index(const std::vector<TensorDim>& dims, size_t axis) {
+  if (dims.size() < 1) {
+    // Prevents empty dim list from causing segfault during program build
+    throw std::runtime_error("index() must be provided non-empty dims list");
+  }
```

Comment:
Note that we'll have to do this kind of validation for all language bindings (so `python` also today). Is there a way we can do this check behind the ffi? I'm thinking in `ast_ops.cc`